src/backend/activity.nim

```diff
@@ -266,8 +296,26 @@ proc fromJson*(e: JsonNode, T: typedesc[ActivityEntry]): ActivityEntry {.inline.
     tokenIn:  if e.contains(tokenInField):
                 some(fromJson(e[tokenInField], Token))
               else:
-                none(Token)
+                none(Token),
   )
+  if e.hasKey(senderField) and e[senderField].kind != JNull:
+    var address: eth.Address
+    fromJson(e[senderField], senderField, address)
+    result.sender = some(address)
+  if e.hasKey(recipientField) and e[recipientField].kind != JNull:
+    var address: eth.Address
+    fromJson(e[recipientField], recipientField, address)
+    result.recipient = some(address)
+  if e.hasKey(chainIdOutField) and e[chainIdOutField].kind != JNull:
+    result.chainIdOut = some(fromJson(e[chainIdOutField], ChainId))
+  if e.hasKey(chainIdInField) and e[chainIdInField].kind != JNull:
```

Comment:
nitpick: code piece repeated several times, easy to screw up the field name when copy pasting. Perhaps make it a small function?